sysutils/SSOProxyAD/src/opnsense/mvc/app/library/OPNsense/Auth/SSOProxyAD.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ *    Copyright (C) 2016 gitdevmod@github.com
+ *
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright notice,
+ *       this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ *    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ *    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ *    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+namespace OPNsense\Auth;
+
+use OPNsense\Core\Config;
+
+/**
+ * Class SSOProxyAD connector
+ * @package OPNsense\Auth
+ */
+class SSOProxyAD implements IAuthConnector
+{
+    public static function getType()
+    {
+        return 'ssoproxyad';
+    }
+
+    /**
+     * user friendly description of this authenticator
+     * @return string
+     */
+    public function getDescription()
+    {
+        return gettext("SSO Proxy AD");
+    }
+   /**
+    * set connector properties
+    * @param array $config connection properties
+    */
+    public function setProperties($config)
+    {
+    }
+
+  /**
+   * unused
+   * @return array mixed named list of authentication properties
+   */
+   public function getLastAuthProperties()
+   {
+       return array();
+   }
+
+   public function authenticate($username,$password)
+   {
```

Comment:
it's probably better to always return false here. Because you're not supposed to use your connector with something else then squid, user/pass authentication should always return "unauthenticated" (false)